Mage.Sets/src/mage/cards/t/TheCurseOfFenric.java

```diff
@@ -0,0 +1,175 @@
+import mage.abilities.Ability;
+import mage.abilities.common.SagaAbility;
+import mage.abilities.keyword.DeathtouchAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BecomesCreatureTargetEffect;
+import mage.abilities.effects.common.FightTargetsEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.NamePredicate;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.filter.predicate.permanent.TokenPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.TheCurseOfFenricMutantToken;
+import mage.game.permanent.token.Token;
+import mage.game.permanent.token.TokenImpl;
+import mage.players.Player;
+import mage.target.common.TargetCreaturePermanent;
+import mage.target.Target;
+import mage.target.TargetPermanent;
+import mage.target.targetadjustment.TargetAdjuster;
+import mage.target.targetpointer.EachTargetPointer;
+
+import java.util.UUID;
+
+/**
+ * @author padfoothelix 
+ */
+public final class TheCurseOfFenric extends CardImpl {
+
+    private static final FilterCreaturePermanent nontokenFilter = new FilterCreaturePermanent("nontoken creature");
+    private static final FilterPermanent mutantFilter = new FilterPermanent(SubType.MUTANT, "mutant");
+    private static final FilterCreaturePermanent fenricFilter = new FilterCreaturePermanent("creature named Fenric");
+
+    static {
+        nontokenFilter.add(TokenPredicate.FALSE);
+        fenricFilter.add(new NamePredicate("Fenric"));
+    }
+
+    public TheCurseOfFenric(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{G}{W}");
+        
+        this.subtype.add(SubType.SAGA);
+
+        // (As this Saga enters and after your draw step, add a lore counter. Sacrifice after III.)
+        SagaAbility sagaAbility = new SagaAbility(this);
+
+        // I -- For each player, destroy up to one target creature that player controls. For each creature destroyed this way, its controller creates a 3/3 green Mutant creature token with deathtouch.
+        sagaAbility.addChapterEffect(
+                this, SagaChapter.CHAPTER_I,
+                ability -> {
+                    ability.addEffect(new TheCurseOfFenricDestroyEffect()
+                            .setText("for each player, destroy up to one target creature " +
+                                    "that player controls. For each creature destroyed " +
+                                    "this way, its controller creates a 3/3 green Mutant " +
+                                    "creature with deathtouch")
+                            .setTargetPointer(new EachTargetPointer())
+                    );
+                    ability.setTargetAdjuster(TheCurseOfFenricAdjuster.instance);
```

Comment:
This should be using the new `ForEachPlayerTargetsAdjuster` instead (see 7a3b069d9e63e28b35b625d3cc807a8ed7f13aa4)